src/PKSim.BatchTool/Services/JsonSimulationRunner.cs

```diff
@@ -41,95 +41,64 @@ public Task RunBatch(dynamic parameters)
       {
          string inputFolder = parameters.inputFolder;
          string outputFolder = parameters.outputFolder;
+         string logFileFullPath = parameters.logFileFullPath;
          BatchExportMode exportMode = parameters.exportMode;
+         NotificationType notificationType = parameters.notificationType;
 
          clear();
 
          return Task.Run(() =>
          {
-
-            _logger.AddInSeparator($"Starting batch run: {DateTime.Now.ToIsoFormat()}");
-
-            var inputDirectory = new DirectoryInfo(inputFolder);
-            if (!inputDirectory.Exists)
-               throw new ArgumentException($"Input folder '{inputFolder}' does not exist");
-
-            var allSimulationFiles = inputDirectory.GetFiles(CoreConstants.Filter.JSON_FILTER);
-            if (allSimulationFiles.Length == 0)
-               throw new ArgumentException($"No simulation json file found in '{inputFolder}'");
-
-            var outputDirectory = new DirectoryInfo(outputFolder);
-            if (!outputDirectory.Exists)
+            using (new BatchLoggerDisposer(_logger, logFileFullPath, notificationType))
```

Comment:
This is the guy that starts the process of logging to the file and disposing the file